### PR TITLE
[SMS-410] Ignore more punctuation at end of urls

### PR DESCRIPTION
**Pull request by jorenby**
Now excludes trailing question marks, exclamation points, colons, and semicolons

https://dripcom.atlassian.net/browse/SMS-410

### CODE REVIEW

**Comment by jorenby** [comment]
Required for https://github.com/DripEmail/drip/pull/18905

File: lib/bypass/filter.rb [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
     attr_reader :content, :fragment
 
-    URL_PATTERN = /\bhttps?:\/\/
-      [a-zA-Z0-9\-\._~:\/\?#\[\]@!$&'\(\)\*\+,;=%]+
-      [a-zA-Z0-9\-_~:\/\?#\[\]@!$&\*\+;=%]/x
+    URL_PATTERN = %r{
+      # Start of a word
+      \b
+
+      # Protocol
```
Comment (natebrunette) [inline lib/bypass/filter.rb]: scheme

File: lib/bypass/filter.rb [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+      # Middle characters
+      [a-zA-Z0-9\-\._~:/\?#\[\]@!$&'\(\)\*\+,;=%]+
+
+      # End character
+      [a-zA-Z0-9\-_~/#\[\]@$&\*\+=%]
+    }x.freeze
 
     def initialize(content, options = {})
-      @content = content.to_s.encode("UTF-8")
+      @content = content.to_s.encode('UTF-8')
```
Comment (natebrunette) [inline lib/bypass/filter.rb]: don't we use double quotes for strings?

File: lib/bypass/filter.rb [hunk 3]
```diff
@@ -19,17 +31,17 @@ def to_s
       content
     end
 
-  private
+    private
```
Comment (natebrunette) [inline lib/bypass/filter.rb]: should be outdented

File: lib/bypass/filter.rb (hunk 1 above)
Comment (jorenby) [reply lib/bypass/filter.rb]: Apparently the `//` starts the Authority section. TIL

I'll just dumb it down some.

**Review by natebrunette — APPROVED** [review]
